contrib/freeze.sh

```diff
@@ -1,10 +1,24 @@
 #!/bin/sh
-if [ $# -ne 1 ] ; then
-	echo Usage: $0 filename \(... will output to stdout\)
-	exit 1
+
+MAGIC=MAGIC_TTSIOD_RS
+RSBEP='rsbep -B 255 -D 223 -R 4080'
+
+if [ $# -ne 1 ]; then
+    echo "Usage: $0 filename (... will output to stdout)"
+    exit 1
 fi
-if [ ! -f "$1" ] ; then echo \"$1\" is not a regular file\! ; exit 1 ; fi
+
+if [ ! -f "$1" ]; then
+    echo "'$1' is not a regular file!"
+    exit 1
+fi
+
 # stat -c doesn't work under OS/X
 # SIZE=`stat -c '%s' "$1"`
-SIZE=`ls -l "$1" | awk '{print $5}'`
-{ echo MAGIC_TTSIOD_RS ; echo $SIZE ; cat "$1"; } | rsbep -B 255 -D 223 -R 4080
+SIZE=$(ls -l "$1" | awk '{print $5}')
```

Comment:
When at it, this should say `SIZE="$( LC_ALL=C ls -l "$1" | LC_ALL=C awk '{print $5}' )"`.

Reply:
I didn't like this piece at all but kept it as-is because `# stat -c doesn't work under OS/X`.

What if we rid of `ls` and use `SIZE=$(find -printf '%s')` instead? Idk how portable is it.

If sticking to`ls`, lets rid of `awk`:

```
fsize() {
    set -- $(ls -n "$1")
    echo "$5"
}

SIZE=$(fsize ...)
```

Reply:
I didn't like this piece at all but kept it as-is because `# stat -c doesn't work under OS/X`.

What if we rid of `ls` and use `SIZE=$(find -printf '%s')` instead? Idk how portable is it.

If sticking to`ls`, lets rid of `awk`:

```
fsize() {
    set -- $(LC_ALL=C ls -n "$1")
    echo "$5"
}

SIZE=$(fsize ...)
```

Also, what is it behing LC_ALL=C in **this** particular case? And why repeat it in two places instead of `export LC_ALL=C` in the top?

Reply:
> ```
> fsize() {
>     set -- $(LC_ALL=C ls -n "$1")
>     echo "$5"
> }
> 
> SIZE=$(fsize ...)
> ```

Yes, this is the best! Thanks!

> Also, what is it behing LC_ALL=C in **this** particular case?

See `<device info>` in https://pubs.opengroup.org/onlinepubs/9699919799/utilities/ls.html

Generally when I am the reviewer I never allow any such string related output/input utilities to work without **explicit** LC_ALL. One literally never knows what else POSIX is hiding...

>And why repeat it in two places instead of `export LC_ALL=C` in the top?

We want to keep as much localized messages (e.g. errors) from the used commands as possible.

Reply:
> We want to keep as much localized messages (e.g. errors) from the used commands as possible.

Oops, I missed this point